### PR TITLE
feat!: move peer exchange into a separate package

### PR DESCRIPTION
**Pull request by danisharora099**
## Problem

Peer Exchange uses `@waku/enr` which brings a lot of dependencies such as cryptographic ones. It was thus seen best to move it to a separate package, similar to `@waku/dns`

## Solution

Separate package for `peer-exchange` in the monorepo

### CODE REVIEW

File: packages/core/src/lib/waku_relay/index.ts [hunk 1]
```diff
@@ -6,9 +6,9 @@ import {
 } from "@chainsafe/libp2p-gossipsub";
 import {
   PeerIdStr,
+  SignaturePolicy,
   TopicStr,
-} from "@chainsafe/libp2p-gossipsub/dist/src/types";
-import { SignaturePolicy } from "@chainsafe/libp2p-gossipsub/types";
+} from "@chainsafe/libp2p-gossipsub/types";
```
Comment (danisharora099) [inline packages/core/src/lib/waku_relay/index.ts:11]: not directly related to the scope of the PR but fixed this while I was debugging some circular dependency red herrings

Comment (fryorcraken) [reply packages/core/src/lib/waku_relay/index.ts:11]: should it be `import type` ?

File: packages/peer-exchange/src/waku_peer_exchange.ts [hunk 2]
```diff
@@ -1,33 +1,28 @@
...
-// import all from "it-all";
 import * as lp from "it-length-prefixed";
 import { pipe } from "it-pipe";
 import { Uint8ArrayList } from "uint8arraylist";
 
-import { selectConnection } from "../select_connection";
-import { getPeersForProtocol, selectPeerForProtocol } from "../select_peer";
-
-import { PeerExchangeRPC } from "./peer_exchange_rpc";
+import { PeerExchangeRPC } from "./rpc";
```
Comment (fryorcraken) [inline packages/peer-exchange/src/waku_peer_exchange.ts:22]:
```suggestion
import { PeerExchangeRPC } from "./rpc.js";
```

tech debt: `@waku/core` has a script `./build-scripts/fix-import.js` that automatically adds all the `.js` extensions to imports for ESM.

I am not convinced this was the best way forward so I did not setup this scripts for all other packages.

The issue now is that the missing `.js` can be unspotted until we release and use it in a different project.

Tracked here https://github.com/waku-org/js-waku/issues/1039

**Review by fryorcraken — CHANGES_REQUESTED** [review]
You added an empty file packages/core/src/lib/waku_store/store.test.ts

File: packages/core/src/lib/waku_relay/index.ts (hunk 1 above)
Comment (danisharora099) [reply packages/core/src/lib/waku_relay/index.ts:11]: Just the types of`PeerIdStr` and `TopicStr` can be imported, however, `SignaturePolicy` needs to be imported fully.

File: packages/peer-exchange/src/waku_peer_exchange.ts [hunk 3]
```diff
@@ -71,12 +66,17 @@ export class WakuPeerExchange implements PeerExchange {
...
-        if (!peerInfo.ENR) return;
-        const enr = await ENR.decode(peerInfo.ENR);
-        peerInfos.push({ ENR: enr });
+      const peerInfos = (await Promise.all(enrPromises)).map((enr) => {
+        return {
+          ENR: enr,
+        };
       });
 
       return { peerInfos };
```
Comment (fryorcraken) [inline packages/peer-exchange/src/waku_peer_exchange.ts:82]: You currently have both of worst world:

1. You have two iterations: `for of` and `map`
2. You are using `map` on the result of `await Promise.all` forcing you to wrap it in brackets.

Either make it readable (ie, solve (2)) or compact (solve (1)).

**Review by fryorcraken — COMMENTED** [review]
Let's keep nwaku at v0.13 in the repo.

store.test.ts empty file still present

File: packages/peer-exchange/src/waku_peer_exchange.ts (hunk 3 above)
Comment (danisharora099) [reply packages/peer-exchange/src/waku_peer_exchange.ts]: addressed - lmk if looks good :)

File: packages/tests/tests/peer_exchange.node.spec.ts [hunk 4]
```diff
@@ -1,63 +1,68 @@
...
     !!nwaku1 && nwaku1.stop();
     !!nwaku2 && nwaku2.stop();
     !!nwaku3 && nwaku3.stop();
+
     !!waku && waku.stop().catch((e) => console.log("Waku failed to stop", e));
   });
 
-  it("Queries successfully", async function () {
-    this.timeout(15_000);
+  it.only("Queries successfully", async function () {
```
Comment (fryorcraken) [inline packages/tests/tests/peer_exchange.node.spec.ts:22]: @danisharora099 I am guessing this should have been merged with `it.skip` instead, right?

File: packages/tests/tests/peer_exchange.node.spec.ts [hunk 5]
```diff
@@ -1,63 +1,68 @@
...
     console.log({ enr });
 
-    await nwaku1.start({
+    await nwaku2.start({
       discv5Discovery: true,
-      manualArgs: [
-        `--discv5-bootstrap-node:${(await nwaku2.info()).enrUri}`,
-        `--ports-shift=5`,
-      ],
+      manualArgs: [`--discv5-bootstrap-node:${enr}`, `--discv5-udp-port:9043`],
```
Comment (fryorcraken) [inline packages/tests/tests/peer_exchange.node.spec.ts:42]: Please remove the `manualArgs` and add `discv5BootstrapNode` to the args and include `discv5-udp-port` to the port finder logic in nwaku @danisharora099

**Pull request merged**